### PR TITLE
update test_fin_builder to reflect local_machine option

### PR DESCRIPTION
**Pull request by cderb**

### CODE REVIEW

File: tuna/worker_interface.py [hunk 1]
```diff
@@ -845,13 +844,7 @@ def run(self):
...
         ret = self.step()  # pylint: disable=no-member
         self.logger.info("proc %s step %s", self.gpu_id, ret)
-        if not ret and (self.poll_retries > 0 and self.poll_retries < 120):
-          pass
-        elif not ret and (self.poll_retries == 0 or self.poll_retries >= 120):
-          if self.poll_retries >= 120:
-            self.logger.warning(
-                'Max poll retries number(120) reached, quiting...')
-            return False
+        if not ret:
```
Comment (alexandraBara) [inline tuna/worker_interface.py:847]: I can see how this can make us wait around for a long time

**Review by alexandraBara — APPROVED** [review]

File: tests/test_fin_builder.py [hunk 2]
```diff
@@ -62,32 +57,22 @@ def add_fin_find_compile_job():
...
+    get_jobs = "SELECT count(*) from conv_job where reason='tuna_pytest_fin_builder' and state='new';"
+    cur.execute(get_jobs)
+    res = cur.fetchall()
+    assert (res[0][0] > 0)
+    num_jobs = res[0][0]
 
-    go_fish_run = "{0}/../tuna/go_fish.py --local_machine --fin_steps miopen_find_compile --session_id 1".format(
-        this_path)
-    os.system(go_fish_run)
+  go_fish_run = "{0}/../tuna/go_fish.py --local_machine --fin_steps miopen_find_compile --session_id 1".format(
```
Comment (JehandadKhan) [inline tests/test_fin_builder.py:70]: If we refactor fin, this can just be a function call into fin

**Comment by cderb** [comment]
@JehandadKhan resolved errors, waiting review

**Review by alexandraBara — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -41,25 +41,49 @@ pipeline {
...
+        agent{  label utils.rocmnode("tunatest") }
         steps {
             script {
             utils.finSolvers()
             }
             } 
         }
         stage("fin applicability"){
-	      agent{  label "gfx908" }
+        //init_session called here
```
Comment (JehandadKhan) [inline Jenkinsfile:53]: We can update init_session to take in arch/num_cu and run this stage on the no-gpu node as well

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**